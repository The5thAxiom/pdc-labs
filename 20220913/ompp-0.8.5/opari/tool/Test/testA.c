int main() {
  int i;

#pragma omp barrier

#pragma pomp no instrument

#pragma pomp inst begin(foo)

#pragma omp parallel
      i = 9;

#pragma pomp inst end(foo)

#pragma pomp instrument

#pragma omp barrier
}
