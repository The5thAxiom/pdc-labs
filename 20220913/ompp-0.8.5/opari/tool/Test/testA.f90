      program testA

      !$OMP BARRIER

      !$POMP NOINSTRUMENT

      !$POMP INST &
      !$POMP BEGIN(foo)

      !$OMP PARALLEL
      i = 9
      !$OMP end PARALLEL

      !$POMP INST END(foo)

      !$POMP INSTRUMENT

      !$OMP BARRIER

      end
