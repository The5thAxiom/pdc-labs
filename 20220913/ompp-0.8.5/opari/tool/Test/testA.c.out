#ifdef _POMP
#  undef _POMP
#endif
#define _POMP 200110

#include "testA.c.opari.inc"
#line 1 "testA.c"
int main() {
  int i;

POMP_Barrier_enter(&omp_rd_26);
#line 4 "testA.c"
#pragma omp barrier
POMP_Barrier_exit(&omp_rd_26);
#line 5 "testA.c"

#line 7 "testA.c"

#pragma pomp inst begin(foo)

#pragma omp parallel
      i = 9;

#pragma pomp inst end(foo)

#line 16 "testA.c"

POMP_Barrier_enter(&omp_rd_27);
#line 17 "testA.c"
#pragma omp barrier
POMP_Barrier_exit(&omp_rd_27);
#line 18 "testA.c"
}
